application/dash_application/views/preqc_rna.py

```diff
@@ -0,0 +1,531 @@
+    "rrna-contamination-slider",
+    "passed-filter-reads-slider",
+
+    # Graphs
+    "total-reads",
+    "unique-reads",
+    "reads-per-start-point",
+    "5-to-3-prime-bias",
+    "correct-read-strand",
+    "coding",
+    "dv200",
+    "rin"
+])
+
+RNA_COL = RnaColumn
+PINERY_COL = pinery.column.SampleProvenanceColumn
+INSTRUMENT_COLS = pinery.column.InstrumentWithModelColumn
+RUN_COLS = pinery.column.RunsColumn
+
+special_cols = {
+    "Total Reads (Passed Filter)": "total_reads_pf",
+    "Percent Uniq Reads": "pct_uniq_reads",
+    "Project": "project",
+    "Percent Correct Strand Reads": "pct_correct_strand_reads",
+    "shape": "shape",
+}
+
+# Set points for graph cutoffs
+graph_cutoffs = {
+    "reads_per_start_point": 5,
+    "rrna_contamination": 50,
+    "pf_reads": 0.01
+}
+
+
+initial_colour_col = PINERY_COL.StudyTitle
+initial_shape_col = PINERY_COL.PrepKit
+initial_shapes = get_shapes_for_values(initial_shape_col)
+
+
+shape_or_colour_by = [
+    {"label": "Project", "value": PINERY_COL.StudyTitle},
+    {"label": "Run", "value": PINERY_COL.SequencerRunName},
+    {"label": "Kit", "value": PINERY_COL.PrepKit},
+    {"label": "Tissue Prep", "value": PINERY_COL.TissuePreparation},
+    {"label": "Library Design", "value": PINERY_COL.LibrarySourceTemplateType},
+]
+
+
+def get_rna_data():
+    """
+    Join together all the dataframes needed for graphing:
+      * RNA-SeqQC (where most of the graphed QC data comes from)
+      * Pinery (sample information)
+      * Instruments (to allow filtering by instrument model)
+      * Runs (needed to join Pinery to Instruments)
+    """
+    # Get the RNA-SeqQC data
+    # NB: have to go two layers down to get the actual cache:
+    #  * QCETLCache(): returns an object of all caches
+    #  * QCETLCache().rnaseqqc: returns the items associated with the
+    #  rnaseqqc cache
+    #  * QCETLCache().rnaseqqc.rnaseqqc: returns the DataFrame/cache named
+    #  "rnaseqqc" within the rnaseqqc cache (as some caches like bcl2fastq
+    #  contain multiple DataFrame/caches)
+    rna_df = QCETLCache().rnaseqqc.rnaseqqc
+    # Cast the primary key/join columns to explicit types
+    rna_df = util.df_with_normalized_ius_columns(rna_df, RNA_COL.Run,
+                                                 RNA_COL.Lane, RNA_COL.Barcodes)
+
+    # Calculate percent uniq reads column
+    rna_df[special_cols["Percent Uniq Reads"]] = round(
+        (rna_df[RNA_COL.UniqReads] / rna_df[RNA_COL.TotalReads]) * 100, 1)
+    rna_df[special_cols["Total Reads (Passed Filter)"]] = round(
+        rna_df[RNA_COL.TotalReads] / pow(10, 6), 3)
+    rna_df[special_cols["Percent Correct Strand Reads"]] = round(
+        rna_df[RNA_COL.CorrectStrandReads] * 100
+    )
+
+    # Pull in sample metadata from Pinery. Keep only the columns we care about
+    pinery_samples = util.get_pinery_samples_from_active_projects()
+    pinery_samples = pinery_samples[
+        [
+            "DV200", # FIXME
```

Comment:
FIXME? here and at 119

Reply:
Yes. By all rights it looks like it should be `PINERY_COL.DV200` but that resolves to `"dv200"` and I get a "no column with this name" error. The error doesn't throw when using the literal `"DV200"`. I haven't yet figured it out and it's not mission-critical at this time so I'm keeping the FIXME and it's high on my to-do list